usl_pipeline/usl_lib/usl_lib/readers/config_readers.py

```diff
@@ -21,3 +23,40 @@ def read_rainfall_amounts(rain_fd: TextIO) -> Sequence[float]:
             continue
         entries.append(float(rainfall_line.group(1)))
     return entries
+
+
+def read_key_value_pairs(fd: TextIO) -> dict:
+    """Reads a config file (.txt, .ini, etc) and returns a dictionary of k/v pairs.
+
+    Args:
+      fd: The file containing the key/value pairs
+
+    Returns:
+      A dictionary of the key/values pairs in the file
+    """
+    # Read lines from file first. If the first line is not blank and doesn't look
+    # like a section header, add [DEAULT] section so it is compliant with
+    # what ConfigParser expects
+    lines = fd.readlines()
+    if lines and not lines[0].strip().startswith("["):
+        lines.insert(0, "[DEFAULT]\n")
+
+    # Create a new StringIO object from the modified lines
+    modified_content = io.StringIO("".join(lines))
+
+    config = configparser.ConfigParser()
+    config.read_file(modified_content)
+
+    config_dict = {}
+
+    # Iterate through all sections and add key-value pairs directly to config_dict
+    for section in config.sections():
+        for key, value in config.items(section):
+            config_dict[key] = value
+
+    # If there are no sections, add key-value pairs from the DEFAULT section
+    if not config.sections():
```

Comment:
How would this happen? The '[DEFAULT]' section we added should be picked up in the above loop. I don't think we need this branch? I think we can get away with just

```python
config = configparser.ConfigParser()
config.read_file(itertools.chain(['[DEFAULT]\n'], fd))
return dict(itertools.chain(config.items(section) for section in config.section()))
```
or even
```python
config = configparser.ConfigParser()
config.read_file(itertools.chain(['[DEFAULT]\n'], fd))
return config['DEFAULT']
```
if we know there won't be any sections in the file we're processing since the `config['DEFAULT']` will return a dict-like object already

Reply:
thanks - that simplifies it. was originally trying to support regular ini files with headers but i don't think we'll need it. we will just assume the files do not include the headers